### ISSUE TITLE
[CoE Starter Kit - QUESTION] Upgrade CoE Starter Kit from Inventory Flows to DATA Export feature 

### ISSUE THREAD

**Issue by pamatihyd** [microsoft/coe-starter-kit#5756]
### Does this question already exist in our backlog?

- [X] I have checked and confirm this is a new question.

### What is your question?

Hi Thank you so much for driving us  in exploring CoE tool kit.

We have an environment where we have installed the latest June release of the CoE kit, including the Audit Logs solution.

If we want to switch from Inventory flows to the Data Export feature and get it up and running, we can follow the steps below:

**Configure the dataflows:** Begin by setting up the dataflows in our environment. This data should include information related to Power Platform components such as PVA's Desktop flows, Power Pages, and Solutions.???

**Disable Admin Sync v3 and Inventory flows:** Once the dataflows have been configured and tested successfully,  can we proceed with turning off the Admin Sync v3 and all existing Inventory flows. This will stop the collection and synchronization of inventory data. ?

**Data consistency and redundancy:** 
By Turning off the Admin Sync v3 and Inventory flows should it  cause any data inconsistency, redundancy, or duplicate records in the Dataverse tables?
 The dataflows, if correctly configured, will ensure the data remains consistent and accurate. ?

**Accessing data in Power Platform:** By relying on the Data Export feature and the configured dataflows, you should be able to access all the required data in Power Platform, including data from PVA's Desktop flows, Power Pages, and Solutions.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.9

### What method are you using to get inventory and telemetry?

Data Export

**Comment by Jenefer-Monroe** [comment]
To convert from sync flow architecture to data export, you should use the Setup Wizard. It will have a screen that asks you which architecture you want to use. Changing to Data Export will move you through the rest of the steps as needed. 

That said, do be sure you turn on the Data Export feature and wait until its filled before doing this move as it can take a week or so for the Data Export product feature to complete its initialization.

**Comment by manuelap-msft** [comment]
Hello,

to answer this question:
`By relying on the Data Export feature and the configured dataflows, you should be able to access all the required data in Power Platform, including data from PVA's Desktop flows, Power Pages, and Solutions.`

Currently, the Data Export feature only includes information on Apps (canvas apps, model driven apps, custom pages) and Cloud Flows.  PVA's Desktop flows, Power Pages, and Solutions. are not included in the inventory provided by Data Export, and the CoE kit continues to gather this data via the sync flows we have today. If you use the Setup Wizard, the right flows will be turned on/off to gather all inventory either from Data Export or via sync flows.

**Comment by pamatihyd** [comment]
Hi thank you we are decided to switch from regular CoE Flows to Data Export Feature.

As per your suggestion we will use the CoE upgrade Wizard only .
But we have one concern 
Now in our CoE Production environment already is having data in the database since the CoE set up 7 months back.
when we switch to DataExport feature , while dataflows are running the data , will it  avoid duplicate data 

I.e I mean will there any duplicate data or simply will get the latest data which is not there in CoE Tables ?

**Comment by Jenefer-Monroe** [comment]
Yes there is no upgrade issue, we just recognize the object (app/flow) to be already in inventory. 
You may want to install the inventory in a test envt first, just to make sure your inventory is complete (meaning that your Data Export has all evnts ready) and then upgrade once you validate that. Many people do this.

**Comment by pamatihyd** [comment]
From this step of wizard , redirecting to Dataflows , but here when we publish the dataflow 
 will it run and skip already loaded data and import only new ?


![image](https://github.com/microsoft/coe-starter-kit/assets/29652263/42454f04-3d06-4296-a89c-6025860c6fd5)

![image](https://github.com/microsoft/coe-starter-kit/assets/29652263/39897ede-69a9-4efa-a1b4-eb6b560c0158)

**Comment by Jenefer-Monroe** [comment]
The process needs to run end to end, you can't run individual pieces of it and expect it to work, this is due to a bunch of complexity. 
In this case I believe we have a flow which runs at the end and calls an orphan clean up process for these as the dataflow is not able to recognize some user types (like app registration). 
Please do your setup in a test envt using the setup wizard to install end to end and then see if you have remaining questions.

**Comment by pamatihyd** [comment]
Thank you so much quick response , 

I am trying to install in test environment only as per your suggestion, I am also not expecting it to work sorry If   I am , sorry trying to cautious and dont want to cause any issue with duplicate . 

I have ran the flow as well 

CoE BYODL - When Maker dataflow refresh is complete.

The only thing I am checking if one table is perfectly loaded then I can work with all other tables .

Fine thanks alot for quick responses 

I will try to complete the whole process and then come back if have any questions

**Comment by pamatihyd** [comment]
as suggested I have configured but its at the end showing the following error message 

![image](https://github.com/microsoft/coe-starter-kit/assets/29652263/812f851d-3f47-4d0b-82bf-f5b17d5b20f1)

**Comment by Jenefer-Monroe** [comment]
I'm afraid helping you set up here in GitHub will be quite difficult. The steps do work so likely the values you entered to the dataflows were incorrect. 
Go look at the data flows and see which one failed. And what the error is. That will help point you.

**Comment by pamatihyd** [comment]
hi I am unable to figure out why dataflows are failing when I upgrade from sync flow architecture to data export,

Environment Dataflow 

}","{""message"":""A record with the specified key values does not exist in admin_maker entity"",""cdsErrorCode"":""0x80060891"",""HelpLink"":""http://go.microsoft.com/fwlink/?LinkID=398563&error=Microsoft.Crm.CrmException%3a80060891&client=platform"",""InnerErrorMessage"":""A record with the specified key values does not exist in admin_maker entity"",""cdsRequestId"":""304b2c6d-5bc4-4abc-9938-181a85e6bd35"",""cdsRequestUtcTime"":""6/22/2023 9:09:09 AM""}"

B y looking at the error I have understood , there is a dependency with makers and Environments Table and I also figured out maker who created environment is still exists in the Makers table , but the Maker GUID is not matching , 

In table , maker GUID for Sytem account is different but when through dataflows it is passing diffrent GUID and its failing all the time since the GUID fields are no editable , I am not able to proceed futher ,

I also tried to skip all the environments with type Developer and Support these environments only failing.

Could you please help ?

**Comment by Jenefer-Monroe** [comment]
To switch back, you would walk through the setup wizard again and select sync flows this time.
Then you would go and turn off auto-refresh on the maker dataflow

**Comment by Jenefer-Monroe** [comment]
closing out as no further action for starter kit team
